integration/docker_integration_test_executor.go

```diff
@@ -80,16 +81,32 @@ func givenThisContainer(t *testing.T, bundle IntegrationTestDockerExecRunnerBund
 		t.Fatalf("")
 	}
 
+	params := []string{"run", "--detach", "-v", localPiper + ":/piper", "--name=" + testRunner.ContainerName}
+	if testRunner.User != "" {
+		params = append(params, fmt.Sprintf("--user=%s", testRunner.User))
+	}
+	if len(bundle.TestDir) > 0 {
+		projectDir := path.Join(wd, path.Join(bundle.TestDir...))
+		params = append(params, "-v", fmt.Sprintf("%s:/project", projectDir))
+	}
+	if len(testRunner.Environment) > 0 {
+		for envVarName, envVarValue := range testRunner.Environment {
+			params = append(params, "--env", fmt.Sprintf("%s='%s'", envVarName, envVarValue))
+		}
+	}
+	params = append(params, testRunner.Image, "sleep", "2000")
+
 	//todo mounts
-	//todo env (secrets)
-	err = testRunner.Runner.RunExecutable("docker", "run", "-d", "-u="+testRunner.User,
-		"-v", localPiper+":/piper", "-v", tempDir+":/project",
-		"--name="+testRunner.ContainerName,
-		testRunner.Image,
-		"sleep", "2000")
+	err = testRunner.Runner.RunExecutable("docker", params...)
 	if err != nil {
 		t.Fatalf("Starting test container has failed %s", err)
 	}
+
+	err = testRunner.Runner.RunExecutable("docker", "exec", "-u=root", testRunner.ContainerName, "chown", "-R", testRunner.User, "/project")
+	if err != nil {
+		t.Fatalf("Chown /project has failed %s", err)
+	}
```

Comment:
This should only happen `if len(bundle.TestDir) > 0`, no?

Reply:
correct, thanks